ovs/dal/hybrids/vdisk.py

```diff
@@ -134,45 +135,50 @@ def _dtl_status(self):
             return sd_status
         return 'checkup_required'
 
-    def _snapshots(self):
+    def _snapshot_ids(self):
         """
-        Fetches a list of Snapshots for the vDisk
+        Fetches the snapshot IDs for this vDisk
         """
-        snapshots = []
-        if self.volume_id and self.vpool:
-            volume_id = str(self.volume_id)
-            voldrv_snapshots = []
+        if not self.volume_id or not self.vpool:
+            return []
+
+        volume_id = str(self.volume_id)
+        try:
             try:
-                try:
-                    voldrv_snapshots = self.storagedriver_client.list_snapshots(volume_id, req_timeout_secs=2)
-                except VolumeRestartInProgressException:
-                    time.sleep(0.5)
-                    voldrv_snapshots = self.storagedriver_client.list_snapshots(volume_id, req_timeout_secs=2)
-            except:
-                pass
+                return self.storagedriver_client.list_snapshots(volume_id, req_timeout_secs=2)
+            except VolumeRestartInProgressException:
+                time.sleep(0.5)
+                return self.storagedriver_client.list_snapshots(volume_id, req_timeout_secs=2)
+        except:
+            return []
 
-            for snap_id in voldrv_snapshots:
-                snapshot = self.storagedriver_client.info_snapshot(volume_id, snap_id, req_timeout_secs=2)
-                if snapshot.metadata:
-                    metadata = pickle.loads(snapshot.metadata)
-                    if isinstance(metadata, dict):
-                        snapshots.append({'guid': snap_id,
-                                          'timestamp': metadata['timestamp'],
-                                          'label': metadata['label'],
-                                          'is_consistent': metadata['is_consistent'],
-                                          'is_automatic': metadata.get('is_automatic', True),
-                                          'is_sticky': metadata.get('is_sticky', False),
-                                          'in_backend': snapshot.in_backend,
-                                          'stored': int(snapshot.stored)})
-                else:
+    def _snapshots(self):
+        """
+        Fetches the information of all snapshots for this vDisk
+        """
+        snapshots = []
+        for snap_id in self.snapshot_ids:
```

Comment:
Better call the internal `self._snapshot_ids()` and put the `info_snapshot(...)` in a try/except.